types/braintree/index.d.ts

```diff
@@ -197,7 +197,7 @@ declare namespace braintree {
         paymentMethodNonce: T extends PaymentMethodNonce ? PaymentMethodNonce : never;
         settlementBatchSumary: T extends SettlementBatchSummary ? SettlementBatchSummary : never;
         subscription: T extends Subscription ? Subscription : never;
-        transaction: T extends Transaction ? Transaction : never;
+        transaction?: T extends Transaction ? Transaction : never;
```

Comment:
This is a breaking change, even for people who are already writing perfectly safe code here:

```ts
if (result.success) {
  doSomethingWith(result.transaction)
}
```

It sure looks like this should be split up into a discriminated union of error response and success response.